### PR TITLE
PIMS-2: Transition from Not in SPL -> In ERP

### PR DESCRIPTION
**Pull request by jdtoombs**
In this PR:

- when a project has the status of `Not in SPL` an option will appear on the ERP tab allowing the user to switch the project status back to `In ERP`
- after clicking this option the user will be prompted with a modal asking them if they would like so send a notification for this status transition
updated_transition

### CODE REVIEW

File: backend/dal/Services/Concrete/ProjectService.cs
Comment (jdtoombs) [inline backend/dal/Services/Concrete/ProjectService.cs:830]: This is what I am not really a fan of with the toggle being on the project

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx
Comment (jdtoombs) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:83]: Section only appears for projects with status `Not in SPL`

File: backend/dal/Migrations/20210702193135_v01.10.07.cs
Comment (Fosol) [inline backend/dal/Migrations/20210702193135_v01.10.07.cs:12]: We don't want to modify the table, as this would stop all future notifications for this project.  All we really need is an option in the API endpoint and the service layer so that on specific requests it will avoid sending notifications.

File: backend/dal/Migrations/v01.10.07/Up/PostUp/01-ProjectStatusTransitions.sql
Comment (Fosol) [inline backend/dal/Migrations/v01.10.07/Up/PostUp/01-ProjectStatusTransitions.sql:3]: You will need a rollback script to remove this transition.

File: backend/dal/Services/Concrete/ProjectService.cs
Comment (Fosol) [reply backend/dal/Services/Concrete/ProjectService.cs:830]: Instead of this, add an options property for the `SetStatusAsync(...)` function that allow for controlling whether notifications are sent.  Then wrap this `SendNotificationsAsync(...)` with an `if (options.SendNotifications) {...}`

File: backend/dal/Services/Concrete/ProjectService.cs
Comment (Fosol) [inline backend/dal/Services/Concrete/ProjectService.cs:959]: It doesn't make sense to have a function called `SendNotificationAsync` which then has an option that stops sending notifications.

File: backend/entities/Project.cs
Comment (Fosol) [inline backend/entities/Project.cs:21]: Don't mutate the `Project` to control an edge-case.

**Review by Fosol — CHANGES_REQUESTED** [review]
Good first attempt.  If you my comments don't make sense, reach out to me.

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx
Comment (devinleighsmith) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:87]: @jdtoombs can you confirm that this modal works correctly if it used multiple times in the same session (ie. not in SPL -> in ERP -> not in SPL -> in ERP)

**Comment by devinleighsmith** [comment]
The video seems to show the transition going from not in spl -> erp approved (not in erp)

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx
Comment (jdtoombs) [reply frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:87]: Modal works! Just looking into the 'Approved for ERP' issue right now

**Review by Fosol — APPROVED** [review]
Looks awesome

**Review by Fosol — CHANGES_REQUESTED** [review]
You'll need to update your DB Migration to `v01.11.01` now as my PR was just merged.

**Review by Fosol — APPROVED** [review]
Awesome

**Review by Fosol — CHANGES_REQUESTED** [review]
Wait until my most recent PR is merged so that I can resolve v01.11.00 migration in DEV.

**Review by Fosol — APPROVED** [review]
V01.11.00 has been successfully deployed to DEV.  You can now merge this PR once you get someone else to approve it :D

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**
